vm/src/obj/objtype.rs

```diff
@@ -288,7 +288,7 @@ fn _mro(cls: &PyClassRef) -> Vec<PyClassRef> {
 /// Determines if `obj` actually an instance of `cls`, this doesn't call __instancecheck__, so only
 /// use this if `cls` is known to have not overridden the base __instancecheck__ magic method.
 #[cfg_attr(feature = "flame-it", flame("objtype"))]
-pub fn isinstance(obj: &PyObjectRef, cls: &PyClassRef) -> bool {
+pub fn isinstance(obj: &impl TypeProtocol, cls: &PyClassRef) -> bool {
```

Comment:
Another option to use here might be a template like this:

```rust
pub fn isinstance<T>(obj: &T, cls: &PyClassRef) -> bool where T: TypeProtocol {
    issubclass(&obj.class(), &cls)
}
```

Did you consider both options? I think the templated version will result in slightly more code, but also faster code.

Reply:
It is a template, I used `impl Trait` and not `dyn Trait`: https://doc.rust-lang.org/edition-guide/rust-2018/trait-system/impl-trait-for-returning-complex-types-with-ease.html

Reply:
I do agree that it's more clear what's happening using explicit generics; `&impl Trait` is definitely confusing.